### PR TITLE
[Hardwaresecuritymodules] Bump up sdk and `az dedicated-hsm`: add parameters `--mgmt-network-subnet` and `--mgmt-network-interfaces`

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Hardwaresecuritymodules

File: src/hardware-security-modules/setup.py [hunk 1]
```diff
@@ -9,9 +9,12 @@
 from codecs import open
 from setuptools import setup, find_packages
 
-# TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
 VERSION = '0.1.0'
```
Comment (jsntcy) [inline src/hardware-security-modules/setup.py:13]: Please change the version to `0.2.0`

File: src/hardware-security-modules/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.2.0
+++++++
+* 'az dedicated-hsm': add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces'
+* Bump up SDK to 2021_11_30
```
Comment (jsntcy) [inline src/hardware-security-modules/HISTORY.rst:8]:
```suggestion
* 'az dedicated-hsm': Add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces'
```

File: src/hardware-security-modules/setup.py (hunk 1 above)
Comment (jsntcy) [inline src/hardware-security-modules/setup.py:13]:
```suggestion
VERSION = '0.2.0'
```

File: src/hardware-security-modules/HISTORY.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Release History
 ===============
 0.2.0
 ++++++
-* 'az dedicated-hsm': add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces' and
+* 'az dedicated-hsm': add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces'
 * Bump up SDK to 2021_11_30
```
Comment (jsntcy) [inline src/hardware-security-modules/HISTORY.rst]: Remove the note of this line as API version used in SDK is an implementation detail that should not be exposed to customers.

File: src/hardware-security-modules/HISTORY.rst [hunk 4]
```diff
@@ -4,7 +4,7 @@ Release History
 ===============
 0.2.0
 ++++++
-* 'az dedicated-hsm': add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces' and
+* 'az dedicated-hsm': add parameters '--mgmt-network-subnet' and '--mgmt-network-interfaces'
```
Comment (jsntcy) [inline src/hardware-security-modules/HISTORY.rst]: Capitalize 'add` to Add' as it's our convention.

File: src/hardware-security-modules/setup.py [hunk 5]
```diff
@@ -27,23 +30,25 @@
...
+except ImportError:
+    pass
+
 with open('README.md', 'r', encoding='utf-8') as f:
     README = f.read()
 with open('HISTORY.rst', 'r', encoding='utf-8') as f:
     HISTORY = f.read()
 
 setup(
-    name='hardware-security-modules',
```
Comment (kairu-ms) [inline src/hardware-security-modules/setup.py:39]: Change this name will cause breaking change, could you use the previous name

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**